QRCoder/QRCodeGenerator.cs

```diff
@@ -405,6 +515,46 @@ void WriteEccLevelAndVersion()
             // Insert the 3-bit mask version directly after the error correction level bits.
             DecToBin(maskVersion, 3, fStrEcc, 2);
         }
+
+        void WriteMicroEccLevelAndVersion()
+        {
+            switch (version)
+            {
+                case -1: // M1
+                    break;
+                case -2: // M2
+                    fStrEcc[level == ECCLevel.L ? 2 : 1] = true; // 001 for L and 010 for M
+                    break;
+                case -3: // M3
+                    if (level == ECCLevel.L)
+                    {
+                        fStrEcc[1] = true; // 011 for L
+                        fStrEcc[2] = true;
+                    }
+                    else
+                        fStrEcc[0] = true; // 100 for M
+                    break;
+                default: // M4
+                    fStrEcc[0] = true;
+                    if (level == ECCLevel.L) // 101 for L
+                        fStrEcc[2] = true;
+                    else if (level == ECCLevel.M) // 110 for M
+                        fStrEcc[1] = true;
+                    else // 111 for Q
+                    {
+                        fStrEcc[1] = true;
+                        fStrEcc[2] = true;
+                    }
+                    break;
+            }
+
+            // Insert the 2-bit mask version directly after the version / error correction level bits.
+            var microMaskVersion =
+                maskVersion == 1 ? 0 :
+                maskVersion == 4 ? 1 :
+                maskVersion == 6 ? 2 : 3;
```

Comment:
```suggestion
            int microMaskVersion = maskVersion switch
            {
                1 => 0,
                4 => 1,
                6 => 2,
                _ => 3
            };
```
I'd alos use pattern matching here.

And it's IMO better readable.